### ISSUE TITLE
Sonoff TH16

### ISSUE THREAD

**Issue by bachoo786** [KmanOz/KmanSonoff#18]
Hi

Is your code compatible with Sonoff TH16? I have uploaded the correct code of yours but I am not registering any temperature and humidity. I have also installed DHT.h library to no joy.

Is there something I need to enable? am I missing anything?

Thanks.

**Comment by AnchorIT** [comment]
That is purely a copy of the original

It definitely works with DTH22 for me, make sure you have the defines set in  <https://github.com/KmanOz/KmanSonoff/commit/7b0f96bd939764f76b502813008436dbba71bf28> config_mc.h

 

 

See his readme

https://github.com/KmanOz/KmanSonoff/blob/master/README.md

 

 <https://github.com/KmanOz/KmanSonoff> KmanSonoff/ <https://github.com/KmanOz/KmanSonoff/tree/master/arduino> arduino/KmanSonoff_v1.00sc/

**Comment by bachoo786** [comment]
Hi for the temperature sensor is it mc or school arduino file? I used sc as it contains the dht22 library and talks about temperature? 

Thanks

**Comment by bachoo786** [comment]
I meant mc.h or sc.h?

**Comment by AnchorIT** [comment]
config_sc.h
change #define NONE  to  #define TEMP
#define DHTTYPE          DHT22

**Comment by bachoo786** [comment]
right so I done the changes above and have subscribed to the topic as well but yet I cannot receive any temperature values. I can turn the switch on and off.

**Comment by bachoo786** [comment]
Quick question : does the temperature sensor only work when the sonoff is connected to the mains? Because I am m testing it out from a USB 3v supply

**Comment by mjoshd** [comment]
Yes. Try connecting it to the mains (after disconnecting from your programming cable of course). I had a similar issue, tried almost everything and thought I was losing my mind, but connecting to mains power solved it and the readings come through just fine now.

**Comment by bachoo786** [comment]
@mjoshd thanks for the reply.

quick question does this Kman’s code work with DS18B20? because i am not getting any readings with DS18B20 but getting readings with AM2301.

**Comment by mjoshd** [comment]
Pretty sure it only works with DHT11 & DHT22. I'd highly recommend using a
DHT22.

**Comment by bachoo786** [comment]
yeah but sonoff do 2 types of temp sensors i.e. AM2301 and DS18B20 I think AM2301 is DHT22 because it works fine but AM2301 is something else ...
